src/Pact/Eval.hs

```diff
@@ -376,29 +382,29 @@ solveConstraint
   -> Eval e (HM.HashMap Text Ref)
   -> Eval e (HM.HashMap Text Ref)
 solveConstraint info refName (Direct t) _ =
-  evalError info $ "found native reference " ++ show t ++ " while resolving module contraints: " ++ show refName
+  evalError info $ "found native reference " <> pretty t <> " while resolving module contraints: " <> pretty refName
 solveConstraint info refName (Ref t) evalMap = do
   em <- evalMap
   case HM.lookup refName em of
     Nothing ->
       case t of
         TConst{..} -> evalMap
-        _ -> evalError info $ "found unimplemented member while resolving model constraints: " ++ show refName
+        _ -> evalError info $ "found unimplemented member while resolving model constraints: " <> pretty refName
     Just (Direct s) ->
-      evalError info $ "found native reference " ++ show s ++ " while resolving module contraints: " ++ show t
+      evalError info $ "found native reference " <> pretty s <> " while resolving module contraints: " <> pretty t
     Just (Ref s) ->
       case (t, s) of
         (TDef (Def _n _mn dt (FunType args rty) _ m _) _,
           TDef (Def _n' _mn' dt' (FunType args' rty') _ _ _) _) -> do
-          when (dt /= dt') $ evalError info $ "deftypes mismatching: " ++ show dt ++ "\n" ++ show dt'
-          when (rty /= rty') $ evalError info $ "return types mismatching: " ++ show rty ++ "\n" ++ show rty'
-          when (length args /= length args') $ evalError info $ "mismatching argument lists: " ++ show args ++ "\n" ++ show args'
+          when (dt /= dt') $ evalError info $ "deftypes mismatching: " <> pretty dt <> line <> pretty dt'
```

Comment:
We should probably drop `line` here, I didn't catch it before but Pact errors generally stick to one line. Not a big deal of course and open to suggestions